### PR TITLE
Cleanup frontend question/dashboard routes

### LINKED ISSUE #1157: Change saved questions page url from "/card/" to something like "/questions"
Just noticed the saved question list is at /card. Seems like I'd expect the url to be /questions given the link title.

### PR DESCRIPTION
**Pull request by tlrobinson**
Sherpa: @tlrobinson

@metabase/core-developers Any objections to this? I think the URL is the only user-facing place we use the "card" terminology. This also matches the public/embed URLs.

* Move `/q` to `/question`
* Move `/card/:cardId` to `/question/:cardId`
* Move `/dash/:dashId` to `/dashboard/:dashId`

TODO:

* [x] frontend
* [x] backend
  * [x] pulses

Resolves #1157

### CODE REVIEW

File: frontend/src/metabase/lib/card.js [hunk 1]
```diff
@@ -24,10 +25,10 @@ export function startNewCard(type, databaseId, tableId) {
 }
 
 // load a card either by ID or from a base64 serialization.  if both are present then they are merged, which the serialized version taking precedence
+// TODO: move to redux
```
Comment (salsakran) [inline frontend/src/metabase/lib/card.js]: Do we actually want to do this? =)

File: frontend/src/metabase/query_builder/actions.js [hunk 2]
```diff
@@ -139,6 +139,7 @@ export const initializeQB = createThunkAction(INITIALIZE_QB, (location, params)
             return { uiControls };
         }
 
+
```
Comment (salsakran) [inline frontend/src/metabase/query_builder/actions.js]: stray newline?

**Review by salsakran — APPROVED** [review]
Works for me. 

This will break old pulses and bookmarks, etc,  so this should going in master/a new major release (0.24) vs 0.23.1 makes sense 👍

**Comment by tlrobinson** [comment]
There are redirects for the old style URLs so it won't break old URLs. But regardless I think this should go in 0.24.

Also, that reminds me I need to change the Pulses URL formatting.

**Comment by tlrobinson** [comment]
@salsakran @camsaul Does the backend emit any other frontend URLs?

**Comment by camsaul** [comment]
@tlrobinson you should update `src/metabase/util/urls.clj` because there's a few utility functions there that need to be tweaked for this change

**Review by camsaul — CHANGES_REQUESTED** [review]
update `src/metabase/util/urls.clj` because there's a few utility functions there that need to be tweaked for this change

**Review by kdoh — APPROVED** [review]

**Comment by tlrobinson** [comment]
Updated the backend.

File: src/metabase/util/urls.clj [hunk 3]
```diff
@@ -20,14 +20,14 @@
 
      (dashboard-url 10) -> \"http://localhost:3000/dash/10\""
```
Comment (camsaul) [inline src/metabase/util/urls.clj]: mind updating the docstring too? :D

File: src/metabase/util/urls.clj [hunk 4]
```diff
@@ -20,14 +20,14 @@
...
   [^Integer id]
-  (format "%s/dash/%d" (public-settings/site-url) id))
+  (format "%s/dashboard/%d" (public-settings/site-url) id))
 
 (defn card-url
   "Return an appropriate URL for a `Card` with ID.
 
      (card-url 10) -> \"http://localhost:3000/card/10\""
```
Comment (camsaul) [inline src/metabase/util/urls.clj]: docstring

**Review by camsaul — CHANGES_REQUESTED** [review]
looks like there's a couple docstrings in the backend that weren't updated

after those are fixed this LGTM

File: src/metabase/util/urls.clj (hunk 4 above)
Comment (tlrobinson) [inline src/metabase/util/urls.clj]: done

File: src/metabase/util/urls.clj (hunk 3 above)
Comment (tlrobinson) [inline src/metabase/util/urls.clj]: done

**Review by camsaul — COMMENTED** [review]
great LGTM

**Review by camsaul — APPROVED** [review]
great LGTM

**Pull request merged**